frame/dapps-staking/src/fix.rs

```diff
@@ -0,0 +1,175 @@
+            } else {
+                Ledger::<T>::iter_keys()
+            };
+            // star iterating, label to be able to break out from inner loop
+            'staker_iter: for staker in staker_iter {
+                // every iteration adds one more read
+                consumed_weight = consumed_weight.saturating_add(T::DbWeight::get().reads(1));
+                // take iter for inner loop from start or last processed contract
+                let contract_iter = if let Some(last_processed_contract) =
+                    restake_fix.last_processed_staker_contract.clone()
+                {
+                    GeneralStakerInfo::<T>::iter_prefix_from(&staker, last_processed_contract)
+                } else {
+                    GeneralStakerInfo::<T>::iter_prefix(&staker)
+                };
+                for (contract, staking_info) in contract_iter {
+                    log::info!(
+                        "Extracting contract stake data for staker address: {:?}.",
+                        staker
+                    );
+                    // every inner iteration also adds one read
+                    consumed_weight = consumed_weight.saturating_add(T::DbWeight::get().reads(1));
+                    let staked_value = staking_info.latest_staked_value();
+                    // get contract address as Vec<u8>
+                    let contract_address = PrefixIterator::last_raw_key(&contract_iter);
+                    // select staking_info for the contract
+                    let mut contract_stake_info = restake_fix
+                        .contract_staking_info
+                        .entry(contract_address.clone().to_vec())
+                        .or_default();
+
+                    // update total and number of stakers
+                    contract_stake_info.total =
+                        contract_stake_info.total.saturating_add(staked_value);
+                    contract_stake_info.number_of_stakers += 1;
+                    // check for total consumed weight
+                    // store last processed contract and break out if the limit is reached
+                    if consumed_weight >= weight_limit {
+                        log::info!(
+                            ">>> Ledger migration stopped after consuming {:?} weight.",
+                            consumed_weight
+                        );
+                        restake_fix.last_processed_staker_contract =
+                            Some(contract_address.clone().to_vec());
+                        break 'staker_iter;
+                    }
+                }
+
+                // when one staker is processed, store it and clear stored contract
+                restake_fix.last_processed_staker_contract = None;
+                restake_fix.last_fully_processed_staker =
+                    Some(Ledger::<T>::storage_map_final_key(staker));
+            }
+        } else {
```

Comment:
As mentioned in the call, this part isn't optimal because there's no reason why we couldn't execute both steps in a single call.